patches/minecraft/net/minecraft/world/WorldProvider.java.patch

```diff
@@ -82,14 +82,18 @@
 +     *
 +     * @param dim Dimension ID
 +     */
-+    public void setDimension(int dim)
++    public void setDimension(net.minecraft.util.ResourceLocation dimID)
 +    {
-+        this.dimensionId = dim;
++        this.dimensionId = dimID;
 +    }
-+    public int getDimension()
++    public net.minecraft.util.ResourceLocation getDimension()
 +    {
 +        return this.dimensionId;
 +    }
++    public int  getDimensionInt()
++    {
++    	return net.minecraftforge.common.Dimension.REGISTRY.func_82594_a(dimensionId).getDimIntID();
```

Comment:
You can change so that the dimension int is kept in the dimension object instead of in the registry.

Reply:
I'm stupid... If I could like twice I would.

Reply:
And you can keep a fastutil int2objectmap to map int dimensional numbers to dimension types